.circleci/config.yml

```diff
@@ -118,36 +189,37 @@ jobs:
       - run: mix local.hex --force
       - run: mix local.rebar --force
 
-      # - restore_cache:
-      #     keys:
-      #       - v1-mix-cache-{{ checksum "mix.lock" }}
-
+      - restore_cache:
+          keys:
+            - v1-mix-cache-test-compile-{{ checksum "mix.lock" }}-{{ .Branch }}-{{ .Revision }}
+      - run:
+          name: Compile
+          command: mix compile
       - run:
-          name: Integration Tests & Coveralls
+          name: Integration Tests & Coveralls Part Common
           command: |
-            export GIT_COMMIT_DESCRIPTION=$(git log --format=%B -n 1 $CIRCLE_SHA1)
+            # Don't submit coverage report for forks, but let the build succeed
             export SHELL=/bin/bash
-            mix coveralls.post \
-              --umbrella \
-              --trace \
-              --include integration \
-              --include wrappers \
-              --sha $CIRCLE_SHA \
-              --branch $CIRCLE_BRANCH \
-              --message "$GIT_COMMIT_DESCRIPTION"
+            if [[ -z "$COVERALLS_REPO_TOKEN" ]]; then
+              mix coveralls.html --parallel --umbrella --include common --exclude watcher --exclude child_chain --exclude test --trace
```

Comment:
if someone adds a tests and tags it `:integration`, but forgets to tag it `:watcher` or `:common` etc, it would get excluded, correct? Same with `:wrappers`. 

how about we have a fourth run of `mix coveralls`, which does `--exclude common/watcher/child_chain` (all three) but does `--include integration --include wrappers`, to scoop all those tests left without tags? Or maybe we remove `:common` altogether, and have that encompass all of the tests not marked explicitly as `watcher` or `child_chain` with `--exclude watcher --exclude child_chain --include integration --include wrappers`

Reply:
Your proposal would run all the tests again, no?
If you're worried about forgetting (I wouldn't be because we're doing PR's for that) maybe the best approach is to remove integration and wrapper tags altogether. That would force the developer to choose something.

Reply:
ah, so `include` takes precedence, right.

Then out of the two options I think current approach is better and let's not forget to tag ;).

Maybe we'll think of some fool-proof way of tagging later.